pages/extensions/amp_example_preview/util/preview.py

```diff
@@ -0,0 +1,98 @@
+import re
+import json
+
+PREVIEW_START_BEGINNING = '<!-- preview\n'
+PREVIEW_END_TAG = r'<!-- /preview -->'
+PREVIEW_PATTERN = re.compile(PREVIEW_START_BEGINNING + r'(.*?)\n-->\n(.*?)' + PREVIEW_END_TAG, re.DOTALL)
+
+
+class ExamplePreview(object):
+
+    def get_start_tag(self):
+        jsonData = json.dumps(self.__dict__)
+        return PREVIEW_START_BEGINNING + escape_html(jsonData) + '\n-->\n'
+
+    def get_end_tag(self):
+        return PREVIEW_END_TAG
+
+    @staticmethod
+    def from_json(jsonData):
+        """
+        :type jsonData: str
+        """
+        result = ExamplePreview()
+        result.__dict__ = json.loads(unescape_html(jsonData))
+        return result
+
+    @staticmethod
+    def for_attributes(index, mode, playground, source):
+        """
+        :type index: int
+        :type mode: str
+        :type playground: bool
+        :type source: str
+        """
+        result = ExamplePreview()
+        result.index = index
+        result.mode = mode
+        result.playground = playground
+        result.source = source
+        return result
+
+def escape_html(content):
```

Comment:
Can't we use an existing method for this?

Reply:
html module comes with Python 3.2 which is not available with grow.
Found that xml.sax.saxutils escape/unescape will also do what we need.